src/main/java/com/csc530/smtlib/SimpleSolver.java

```diff
@@ -0,0 +1,512 @@
+package com.csc530.smtlib;
+
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+
+import org.smtlib.ICommand;
+import org.smtlib.ICommand.IScript;
+import org.smtlib.ICommand.Ideclare_fun;
+import org.smtlib.ICommand.Ideclare_sort;
+import org.smtlib.ICommand.Idefine_fun;
+import org.smtlib.ICommand.Idefine_sort;
+import org.smtlib.IExpr;
+import org.smtlib.IExpr.IAsIdentifier;
+import org.smtlib.IExpr.IAttribute;
+import org.smtlib.IExpr.IAttributeValue;
+import org.smtlib.IExpr.IAttributedExpr;
+import org.smtlib.IExpr.IBinaryLiteral;
+import org.smtlib.IExpr.IBinding;
+import org.smtlib.IExpr.IDecimal;
+import org.smtlib.IExpr.IDeclaration;
+import org.smtlib.IExpr.IError;
+import org.smtlib.IExpr.IExists;
+import org.smtlib.IExpr.IFcnExpr;
+import org.smtlib.IExpr.IForall;
+import org.smtlib.IExpr.IHexLiteral;
+import org.smtlib.IExpr.IKeyword;
+import org.smtlib.IExpr.ILet;
+import org.smtlib.IExpr.INumeral;
+import org.smtlib.IExpr.IParameterizedIdentifier;
+import org.smtlib.IExpr.IStringLiteral;
+import org.smtlib.IExpr.ISymbol;
+import org.smtlib.ILogic;
+import org.smtlib.IPos;
+import org.smtlib.IResponse;
+import org.smtlib.IResponse.IAssertionsResponse;
+import org.smtlib.IResponse.IAssignmentResponse;
+import org.smtlib.IResponse.IAttributeList;
+import org.smtlib.IResponse.IProofResponse;
+import org.smtlib.IResponse.IUnsatCoreResponse;
+import org.smtlib.IResponse.IValueResponse;
+import org.smtlib.ISolver;
+import org.smtlib.ISort.IAbbreviation;
+import org.smtlib.ISort.IApplication;
+import org.smtlib.ISort.IFamily;
+import org.smtlib.ISort.IFcnSort;
+import org.smtlib.ISort.IParameter;
+import org.smtlib.ITheory;
+import org.smtlib.IVisitor;
+import org.smtlib.IVisitor.VisitorException;
+import org.smtlib.SMT;
+import org.smtlib.SMT.Configuration;
+
+import com.csc530.sat.DecisionDiagram;
+import com.csc530.sat.DecisionDiagramNode;
+import com.csc530.sat.SATVariable;
+
+class SimpleSolver implements ISolver {
```

Comment:
where did the package scope go?
please add one

also comments on public classes / methods are helpful